xarray/namedarray/core.py

```diff
@@ -308,7 +354,13 @@ def chunks(self) -> tuple[tuple[int, ...], ...] | None:
         NamedArray.chunksizes
         xarray.unify_chunks
         """
-        return getattr(self._data, "chunks", None)
+        data = self._data
+        # reveal_type(data)
+        if is_chunked_duck_array(data):
+            # reveal_type(data)
+            return data.chunks
+        else:
+            return None
```

Comment:
```python
        data = self._data
        reveal_type(data)  #  note: Revealed type is "T_DuckArray`1"
        if is_chunked_duck_array(data):
            reveal_type(data)  #  note: Revealed type is "<nothing>"
            return data.chunks # error: <nothing> has no attribute "chunks"  [attr-defined]
        else:
            return None
```
@headtr1ck Do you (or anyone else) understand why `data` isn't narrowed down to `T_ChunkedArray`?

Reply:
Not sure actually.

But the "`1" in the first reveal_type usually is already an indication that something is not correct. Even though I have never figured out why Mypy does this and what it means (somehow the type is not exactly known at this time or part of a Union or something like that).

Reply:
Somehow I like the `getattr` implementation better than the explicit check... it's just much easier to read and understand.

Still your issue indicates that something is wrong somewhere...